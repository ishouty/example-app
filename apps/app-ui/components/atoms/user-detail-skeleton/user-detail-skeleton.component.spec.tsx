import { render, RenderResult, screen } from '@testing-library/react';
import UserDetailSkeleton from './user-detail-skeleton.component';

let component: RenderResult;
describe('UserDetailSkeleton Component', () => {
  beforeEach(() => {
    component = render(<UserDetailSkeleton />);
  });

  test('matches snapshot', () => {
    expect(component.container).toMatchSnapshot();
  });

  test('renders 5 skeleton descriptioon and one avatar', () => {
    const { container } = component;
    const skeletonLines = component.container.querySelectorAll('.space-y-1');

    expect(component.getAllByTestId('userdetail-container')).toHaveLength(1);
    expect(container.querySelectorAll('svg')).toHaveLength(1);
    expect(skeletonLines).toHaveLength(5);
  });
});
