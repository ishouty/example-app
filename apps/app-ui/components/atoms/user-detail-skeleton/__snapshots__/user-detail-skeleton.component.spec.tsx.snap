// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetailSkeleton Component matches snapshot 1`] = `
<div>
  <div
    class="flex items-center p-6 max-w-sm mx-auto bg-white rounded-xl shadow-sm space-x-4 animate-pulse"
    data-testid="userdetail-container"
  >
    <div
      class="shrink-0"
    >
      <svg
        aria-hidden="true"
        class="tabler-icon tabler-icon-user h-12 w-12 text-gray-300"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 7a4 4 0 1 0 8 0a4 4 0 0 0 -8 0"
        />
        <path
          d="M6 21v-2a4 4 0 0 1 4 -4h4a4 4 0 0 1 4 4v2"
        />
      </svg>
    </div>
    <div
      class="text-left space-y-2 flex-1"
    >
      <div
        class="space-y-1"
      >
        <div
          class="h-4 bg-gray-300 rounded w-24"
        />
        <div
          class="h-6 bg-gray-300 rounded w-64"
        />
      </div>
      <div
        class="space-y-1"
      >
        <div
          class="h-4 bg-gray-300 rounded w-24"
        />
        <div
          class="h-6 bg-gray-300 rounded w-64"
        />
      </div>
      <div
        class="space-y-1"
      >
        <div
          class="h-4 bg-gray-300 rounded w-24"
        />
        <div
          class="h-6 bg-gray-300 rounded w-64"
        />
      </div>
      <div
        class="space-y-1"
      >
        <div
          class="h-4 bg-gray-300 rounded w-24"
        />
        <div
          class="h-6 bg-gray-300 rounded w-64"
        />
      </div>
      <div
        class="space-y-1"
      >
        <div
          class="h-4 bg-gray-300 rounded w-24"
        />
        <div
          class="h-6 bg-gray-300 rounded w-64"
        />
      </div>
    </div>
  </div>
</div>
`;
